src/mono/mono/mini/interp/interp.c

```diff
@@ -2162,12 +2164,11 @@ typedef struct {
 } InterpEntryData;
 
 static MONO_ALWAYS_INLINE gboolean
-is_method_multicastdelegate_invoke (MonoMethod *method)
+is_imethod_multicastdelegate_invoke (InterpMethod *imethod)
 {
-	gboolean expected = method->is_invoke;
-	gboolean actual = (m_class_get_parent (method->klass) == mono_defaults.multicastdelegate_class) && !strcmp (method->name, "Invoke");
-	assert (actual == expected)
-	return actual;
+	if (!imethod->is_invoke)
+		return FALSE;
+	return (m_class_get_parent (imethod->method->klass) == mono_defaults.multicastdelegate_class);
```

Comment:
Can't is_invoke be set based on this ?

Reply:
Is multicastdelegate_class always going to be available up front? Couldn't we be running cctors in the interpreter while we initialize types?

Reply:
mono_defaults is initialized very early.

Reply:
Will test it.

Reply:
It works 👍